Option to target frames with named items

omwah opened the pull request:
Add option to configuration that when enabled will allow items in the item frame to be named a certain name for those frames to be target. This is a way to ensure that items are sorted only to the intended item frames. Requires naming the item in the frame, not the frame itself. I couldn't
find a way to target a named item frame.

omwah committed:
Add option to configuration that when enabled will allow items in the
frame to be named a certain name for those frames to be target. This is
a way to ensure that items are sorted only to the intended item frames.
Requires naming the item in the frame, not the frame itself. I couldn't
find a way to target a named item frame.

omwah committed:
Add option to configuration that when enabled will allow items in the
frame to be named a certain name for those frames to be target. This is
a way to ensure that items are sorted only to the intended item frames.
Requires naming the item in the frame, not the frame itself. I couldn't
find a way to target a named item frame.

jhuckaby committed:
Version 2.0.2

- Added new Minecraft 1.16 items.
- Fixed a few minor issues.

p-hennessy commented:
Looked into this as well; item frames don't seem to expose a name attribute on it's NBT data; not sure why, maybe its a bug in the game? Either way this method works well!

isaaclepes commented:
I re-wrote a lot of this datapack to create a version customized for my server.  My tagger function checks for named item in frame and sets tags.  I used command_block to restrict usage to operators, but the item should not matter.  You can even omit the id: part.  No reason you can't use the same check for a teleport command.

execute as @e[type=minecraft:item_frame,tag=!sss_shulker_target,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Shulker Target"}'}}}}] run function sss:targetshulker
execute as @e[type=minecraft:item_frame,tag=!sss_sort_target,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Sort Target"}'}}}}] run function sss:targetsort
execute as @e[type=minecraft:item_frame,tag=!sss_furnace_target,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Furnace Target"}'}}}}] run function sss:targetfurnace
execute as @e[type=minecraft:item_frame,tag=!sss_smelt_target,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Smelt Target"}'}}}}] run function sss:targetsmelt
execute as @e[type=minecraft:item_frame,tag=!sss_smoker_target,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Smoker Target"}'}}}}] run function sss:targetsmoker
execute as @e[type=minecraft:item_frame,tag=!sss_furnace_fuel,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Furnace Fuel"}'}}}}] run function sss:targetfuelfurnace
execute as @e[type=minecraft:item_frame,tag=!sss_smelt_fuel,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Smelt Fuel"}'}}}}] run function sss:targetfuelsmelt
execute as @e[type=minecraft:item_frame,tag=!sss_smoker_fuel,nbt={Item:{id:"minecraft:command_block",tag:{display:{Name:'{"text":"Smoker Fuel"}'}}}}] run function sss:targetfuelsmoker

omwah committed:
Add script to check a new config against an old one to make sure all
blocks are accounted for in the new config. To make sure you didn't make
mistakes when regrouping.

omwah committed:
Add my custom sorting configuration that puts wool and wool carpet in
seperate chests and puts nether update stuff in to their own chest.

omwah committed:
Add Minecraft 1.17 items with 4 new groupings

omwah committed:
Merge branch 'master' of github.com:omwah/magic-sorting-system

omwah committed:
Move smooth basalt

omwah committed:
Update for MC 1.19 items
